### ISSUE TITLE
mAP is always 0 and there is an error running export.py

### ISSUE THREAD

**Issue by tian123-zxc** [ultralytics/yolov5#8753]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

1.    I am using a custom dataset where mAP is always 0 until the end of training.  I also tried lowering the version of cuda and pytorch, but it didn't work
![image](https://user-images.githubusercontent.com/83101407/181415982-9713a1ff-1e09-46b9-a16c-366797ec7d1a.png)
![RRCFKEX{18L4DQ1%ER_Z_W](https://user-images.githubusercontent.com/83101407/181416283-674ab804-c1b7-44d5-a7a1-de4d2285bb94.png)

2.    After the model is trained I try to convert him to onnx model, but he got the following error, how can I modify it.
![image](https://user-images.githubusercontent.com/83101407/181416507-f44d0143-06a8-47f6-9203-fcc424f96e63.png)

**Comment by tian123-zxc** [comment]
My configuration is as follows：
cuda：11.6
torch：1.12.0
win11

**Comment by glenn-jocher** [comment]
@tian123-zxc your val set has no labels, i.e. see it says labels 0 in your console output. You need to fix this in your dataset. 

Your train set appears correct as your train losses seem fine.

**Comment by glenn-jocher** [comment]
@tian123-zxc good news 😃! Your original issue may now be improved ✅ in PR #8782. This PR adds a clear warning when no labels are found and explains that no metrics can be computed.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
